packages/overlays/docs/OverlayController.md

```diff
@@ -0,0 +1,192 @@
+
+OverlayController is the single class we instantiate whenever creating an overlay instance.
+Based on provided config, it will handle:
+
+- DOM position (local vs global)
+- positioning logic
+- accessibility
+- interaction patterns.
+
+and has the following public functions:
+
+- **show()**, to show the overlay.
+- **hide()**, to hide the overlay.
+- **toggle()**, to toggle between show and hide.
+
+All overlays exists of an invokerNode and a contentNode
+
+- **contentNode**, the toggleable content of the overlay
+- **invokerNode**, the element that can trigger showing (and hiding) the overlay. For local overlays, this is the relative element the content is positioned to.
+
+For DOM position, local refers to overlays where the content is positioned next to the invokers they are related to, DOM-wise.
+Global refers to overlays where the content is positioned in a global root node at the bottom of `<body>`.
+
+## Configuration options
+
+In total, we should end up with configuration options as depicted below, for all possible overlays.
+All boolean flags default to 'false'.
+
+```text
+- {Boolean} trapsKeyboardFocus - rotates tab.
+- {Boolean} hidesOnEsc - hides the overlay when pressing [esc].
+- {Element} elementToFocusAfterHide - the element that should be called `.focus()` on after dialog closes.
+- {Boolean} hasBackdrop - whether it should have a backdrop. (local mode only)
+- {Boolean} isBlocking - hides other overlays when multiple are opened.
+- {Boolean} preventsScroll - prevents scrolling body content when overlay opened.
+- {Object} viewportConfig - placementMode: local only
+  - {String} placement: 'top-left' | 'top' | 'top-right' | 'right' | 'bottom-left' |'bottom' | 'bottom-right' | 'left' | 'center'
+- {Object} popperConfig - placementMode: local only
+  - {String} placement: 'top-left' | 'top' | 'top-right' | 'right' | 'bottom-left' |'bottom' | 'bottom-right' | 'left' | 'center'
+```
+
+> Note: popperConfig reflects [Popper.js API](https://popper.js.org/popper-documentation.html)
+
+## Specific Controllers
+
+You can find our existing configurations [here](../src/configurations):
+
+- withModalDialogConfig
+- withDropdownConfig
+- withBottomSheetConfig
+
+You import these using ES Modules, and then simply call them inside your OverlayController instantiation:
+
+```js
+const ctrl = new OverlayController({
+  ...withModalDialogConfig(),
+  invokerNode,
+  contentNode,
+});
+```
+
+## Responsive switching
+
+Currently we support switching between overlay configurations. Keep in mind however:
+
+- We do not yet support switching between overlay configurations while the content is shown. If you try, it will close the content if it is open, and the user will need to re-open. WIP.
+- We DO support switching the positioning configurations (`popperConfig` property for `local` mode, `viewportConfig` for `global` mode) while content is shown.
```

Comment:
I will remove this line, because I found out later that this is still not supported in some cases (through the public config prop, if you change only the placement, but the base config has other keys, it will still fail to update without closing your overlay)